### PR TITLE
Amoreau Open-sourcing JavaScript

### PR DESCRIPTION
**Pull request by amoreauCoveo**
Pull Request for new blogpost about Open-Source JavaScript

### CODE REVIEW

File: _posts/2016-09-27 Open-Sourcing JavaScript.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+Although it is called the JavaScript Search Framework, the code is written using [TypeScript](http://www.typescriptlang.org/). The TypeScript code is then compiled into JavaScript, which allows us to modify our code base without the fear of breaking the framework.
+
+##Why open-sourcing it?
+
+As Coveo continues to grow, we have more and more partners that need to implement our Search UI in a website, and tweak it in different ways to fit their needs.
+
+Before open-sourcing the code, our partners had to do some serious code-gymnastics to bypass certain sections of the Search Framework and implement their own features.
+
+We improved the flexibility and usability of the framework, and by open-sourcing it, we are giving our advanced Coveo partners a means to take their implementation to the next level by allowing them to take the code and play with it to their heart’s content.
```
Comment (sbelzile-Coveo) [inline _posts/2016-09-27 Open-Sourcing JavaScript.md]: a means? => a mean

Comment (amoreauCoveo) [inline _posts/2016-09-27 Open-Sourcing JavaScript.md]: Non, means
https://en.oxforddictionaries.com/definition/means

**Comment by francoislg** [comment]
Rien à ajouter :+1:

:gg:

File: _posts/2016-09-27 Open-Sourcing JavaScript.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+From now on, the documentation is generated using [TypeDoc](https://github.com/TypeStrong/typedoc/), which means that the documentation is made from comments left in the code. To access the current documentation, see the [Coveo Search UI documentation](https://coveo.github.io/search-ui/).
+
+This way, it is much easier for developers to remember to update the documentation when adding new features or modifying existing ones.
+
+This also means that anyone playing with the open-source code is able to see the documentation right next to the component code without having to refer to an external website.
+
+Furthermore, the TypeDoc generated documentation can be easily implemented directly in the UI components, such as the [JavaScript Search Interface Editor](http://www.coveo.com/go?dest=cloudhelp&lcid=9&context=230). This way, the component can be used even by people who are not familiar with coding.
```
Comment (sbelzile-Coveo) [inline _posts/2016-09-27 Open-Sourcing JavaScript.md]: This way, the component [...], puisque tu ne parles pas vraiment d'un component en particulier, j'écrirais either "the components" or "a Coveo component"

Comment (amoreauCoveo) [inline _posts/2016-09-27 Open-Sourcing JavaScript.md]: Yes, components! Thanks!

File: _posts/2016-09-27 Open-Sourcing JavaScript.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+We modified our project to instead use ES6 CommonJS modules and [webpack](https://webpack.github.io/) in order to have a much more flexible bundle. Technically speaking, specifying any entry point in the project results in a coherent bundle that executes correctly at run time.
+
+This also brought some development perks, such as a [webpack dev server](https://webpack.github.io/docs/webpack-dev-server.html), which makes developing with the Search UI much more enjoyable.
+
+##Can people contribute to the project?
+
+We would love for people to contribute to our Search UI project! Simply make a Pull Request, and a feature or improvement that you coded could be added to the Coveo Search Framework.
+
+This not only makes our developers aware of your implementations, but it also makes every other project you work on that much easier to start.
```
Comment (sbelzile-Coveo) [inline _posts/2016-09-27 Open-Sourcing JavaScript.md]: you work on that much easier to start => you work on much easier to start? (pas de that)

File: _posts/2016-09-27 Open-Sourcing JavaScript.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+Coveo is very open to new ideas and features, and would love to hear from what you think should be improved. We await your pull request in our [GitHub Project](https://github.com/coveo/search-ui), and we hope that you enjoy the new Coveo Search Framework!
```
Comment (sbelzile-Coveo) [inline _posts/2016-09-27 Open-Sourcing JavaScript.md]: We await your pull request => pourquoi pas request**s**?

File: _posts/2016-09-27 Open-Sourcing JavaScript.md (hunk 3 above)
Comment (amoreauCoveo) [inline _posts/2016-09-27 Open-Sourcing JavaScript.md]: I'd tend to keep the "that." It has a different meaning.
Saying "that much easier" means it is easier by "that much," whereas saying "much easier to start" is a different assessment of its difficulty.
The first expression means you might skip a few steps, making it easier. The second means it make the thing, as a whole, easier.

File: _posts/2016-09-27 Open-Sourcing JavaScript.md (hunk 4 above)
Comment (amoreauCoveo) [inline _posts/2016-09-27 Open-Sourcing JavaScript.md]: Yes! Requests plural! :P Sorry

File: _posts/2016-09-27 Open-Sourcing JavaScript.md (hunk 3 above)
Comment (sbelzile-Coveo) [inline _posts/2016-09-27 Open-Sourcing JavaScript.md]: Comme tu veux. C'est toi qui connais ça. C'est que, pour moi, quand je dis "that much", c'est que le "that" représente une quantité x. Ce n'est pas le cas dans ta phrase.

Comment (sbelzile-Coveo) [inline _posts/2016-09-27 Open-Sourcing JavaScript.md]: Also, je me trouve téteux, mais "every other": http://dictionary.cambridge.org/dictionary/english/every-other-something,

j'enlèverais le other

Comment (amoreauCoveo) [inline _posts/2016-09-27 Open-Sourcing JavaScript.md]: Ahh mais "every" et "other" sont pas associé de la même façon dans le tree de la phrase. "Other" est pas attaché à every, mais à project.
Still, might rephrase to remove confusion

**Pull request merged**